awx/main/credential_plugins/hashivault.py

```diff
@@ -118,7 +125,11 @@ def handle_auth(**kwargs):
 def approle_auth(**kwargs):
     role_id = kwargs['role_id']
     secret_id = kwargs['secret_id']
-    auth_path = kwargs.get('auth_path') or 'approle'
+    # we first try to use the 'auth_path' from the metadata
+    # if not found we try to fetch the 'approle_auth_path' from inputs
+    # if not found we use the default value 'approle'
+    auth_path = \
+        kwargs.get('auth_path', kwargs.get('approle_auth_path', "approle"))
```

Comment:
```suggestion
        kwargs.get('auth_path', kwargs['default_auth_path'])
```

Reply:
Right, I did not think about the empty string use case, thanks